stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -3201,33 +3197,23 @@ final internal class _Native${Self}StorageKeyNSEnumerator<
 }
 #endif
 
-/// This class is an artifact of the COW implementation.  This class only
-/// exists to keep separate retain counts separate for:
-/// - `${Self}` and `NS${Self}`,
-/// - `${Self}Index`.
-///
-/// This is important because the uniqueness check for COW only cares about
-/// retain counts of the first kind.
-///
-/// Specifically, `${Self}` points to instances of this class.  This class
-/// is also a proper `NS${Self}` subclass, which is returned to Objective-C
-/// during bridging.  `${Self}Index` points directly to
-/// `_Native${Self}Storage`.
-final internal class _Native${Self}StorageOwner<${TypeParametersDecl}>
+#if _runtime(_ObjC)
+/// This class exists for Objective-C bridging. It holds a reference to a
+/// `_Native${Self}Storage`, and can be upcast to `NS${Self}` when
+/// bridging is necessary.
+final internal class _Native${Self}BridgingStorage<${TypeParametersDecl}>
```

Comment:
Sorry for a terse reply, here are some more details that I hope are helpful.

The `Array` does this trick in `ContiguousArrayBuffer.swift`, in `_ContiguousArrayBuffer.init(uninitializedCount: Int, minimumCapacity: Int)`.  It creates an instance of `_ContiguousArrayStorage` (which is a subclass of `_SwiftNativeNSArray`) using `ManagedBufferPointer` that allows it to get the tail-allocation.

In the non-Objective-C runtime we just define an empty `_SwiftNativeNSArrayWithContiguousStorage` class, but still go through `ManagedBufferPointer` to get the tail-allocation.

Does this make sense?